### ISSUE TITLE
[AWS SDK for Go Migration] Outposts Service

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#36195]
### Description
To align with the dates described on #32976 the Outposts Service requires migration to rely fully on [AWS SDK for Go v2](https://github.com/aws/aws-sdk-go-v2). This change should be mechanical in nature and not introduce any behavior changes.

The maintainer team will be working on these migrations but as always, help from the community is always appreciated. We would require passing acceptance tests for the entire set of resources being migrated. (We will also validate on review)

For further information about the migration, please see the parent issue #32976.

### Resources In Scope for Migration
```

```

### Data Sources In Scope for Migration
```
"aws_outposts_asset"
"aws_outposts_assets"
"aws_outposts_outpost"
"aws_outposts_outpost_instance_type"
"aws_outposts_outpost_instance_types"
"aws_outposts_outposts"
"aws_outposts_site"
"aws_outposts_sites"
```

### References

- #32976

**Comment by breathingdust** [comment]
Closed via #38075

**Comment by DanielRieske** [comment]
@breathingdust Can we re-open this issue? This service hasn't been migrated yet, the linked PR were EC2 Outposts resources.

I will however pick this issue up in the upcoming days
